project-structure.md

```diff
@@ -63,12 +67,28 @@ What you must do:
 - Name function arguments exactly as they are named in the top-level modules
   - You do not need to implement all arguments, but you must name them identically
   - All arguments must be keyword arguments (except the primary arguments)
+  - If an argument's default value should be read from `config`, rather than hardcoded, set its
+    default value to the *data type* `ConfigVar` (see below)
   - If you need to add new arguments,
         feel free, but MAKE SURE you add the argument to the top-level function, as a keyword arg.
         This is the only time you should need to modify the top-level functions while localizing.
         Ensure that any new arguments are at the end of the function signatures, both in the
         top-level function, and in your localized function.
 
+If a function argument's default value should be read from config, it should be set
+to the *data type* `ConfigVar` (not an instance.) For example, here is `parse.extract_number()`'s
+top-level signature:
```

Comment:
What's the benefit of introducing this specific datatype compared to using something generic like `None`?
It makes it more explicit in the signature that it's a config variable as opposed to "not used". Though we could also document that in the docstrings.

I'm not taking a position on what's best, just curious on the thought process behind it.

Reply:
It's unique, so the wrapper can rely on it to mean what it means. No fancy magic to determine when a value needs to be read from config, just, `if param is ConfigVar`. The old business with "what if it's positional" takes more code =P